tensorflow_lite_support/cc/task/vision/utils/image_tensor_specs.cc

```diff
@@ -178,9 +178,32 @@ StatusOr<ImageTensorSpecs> BuildInputImageTensorSpecs(
   // The expected layout is BHWD, i.e. batch x height x width x color
   // See https://www.tensorflow.org/guide/tensors
   const int batch = input_tensor->dims->data[0];
-  const int height = input_tensor->dims->data[1];
-  const int width = input_tensor->dims->data[2];
+  int height = input_tensor->dims->data[1];
+  int width = input_tensor->dims->data[2];
   const int depth = input_tensor->dims->data[3];
+  // Determine if the input shape is resizable.
+  const TfLiteIntArray* dims_signature = input_tensor->dims_signature;
+
+  if (dims_signature != nullptr)
+  {
+      if (dims_signature->data[3] == -1 || dims_signature->data[0] == -1)
+        return CreateStatusWithPayload(StatusCode::kInvalidArgument,
+                                       "Only height and width is mutable for now.",
+                                       TfLiteSupportStatus::kInvalidArgumentError);
+      
+      const int height_mutable = dims_signature->data[1];
+      const int width_mutable = dims_signature->data[2];
+
+      if (height_mutable || width_mutable)
+      {
+        // How do I get the image height and width???
+        height = height_mutable ? height : ???;
+        width = width_mutable ? width : ???;
```

Comment:
Since it is dynamic, how do I get image height and width? Do I need additional args in ```BuildInputTensorSpecs``` ? @lu-wang-g @khanhlvg

Reply:
We could get image height and width from the ```const FrameBuffer& frame_buffer``` arg, but then ```BuildInputTensorSpecs``` doesn't have access to it. Should I move this code to ```Preprocess()```?

Reply:
Didn't see your questions here first. But yes, I think you should move the code to Preprocess() instead.